core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -168,6 +168,13 @@ class KafkaApis(val requestChannel: RequestChannel,
     val response =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+        // Clearing out the cache for groups that belong to an offsets topic partition for which this broker was the leader,
+        // since this broker is no more a replica for that offsets topic partition
```

Comment:
@hachikuji : This is done to protect for a case where a broker was a leader earlier and on reassignment it was moved out of the replica list and a new leader is elected from the remaining replicas in ISR.  
The current comment tries to say the above. 
Sorry I might not be understanding correctly but are you suggesting we give an example in the comments? I can do add it if that is what you are hinting at.